internal/service/flexcluster/resource_test.go

```diff
@@ -136,11 +144,13 @@ func checksFlexCluster(projectID, clusterName string, terminationProtectionEnabl
 		"version_release_system",
 		"provider_settings.provider_name",
 	}
-	checks = acc.AddAttrChecks(resourceName, checks, attrMap)
-	checks = acc.AddAttrChecks(dataSourceName, checks, attrMap)
-	checks = acc.AddAttrSetChecks(resourceName, checks, attrSet...)
-	checks = acc.AddAttrSetChecks(dataSourceName, checks, attrSet...)
-	return resource.ComposeAggregateTestCheckFunc(checks...)
+	checks = acc.AddAttrChecks(dataSourcePluralName, checks, pluralMap)
+	for i := 0; i < flexClusterCount; i++ {
+		checks = acc.AddAttrSetChecks(resourceName, checks, "name", "termination_protection_enabled")
+		checks = acc.AddAttrSetChecks(dataSourceName, checks, "id")
+		checks = acc.AddAttrSetChecks(dataSourcePluralName, checks, "results.0.id")
+	}
```

Comment:
Can these additional checks be handled as part of the `attrSet` that is passed to `CheckRSAndDS`?

Reply:
Yes agreed, initial motivation was to iterate over and check all flex Clusters in pluralDS but I see now that this implementation does not even achieve that and testing only includes one flex cluster as is. Removed 👍